File: Store.Web/src/app/models/Pizza.decorator.ts
import { IPizza } from "./IPizza";

export abstract class PizzaDecorator implements IPizza {
    protected pizza: IPizza;

    constructor(public inputPizza: IPizza) {
        this.pizza = inputPizza;
    }

    GetIngredients(): string {
        return this.pizza.GetIngredients();
    }

    GetCost(): number {
        return this.pizza.GetCost();
    }

    GetImageUrl(): string {
        return this.pizza.GetImageUrl();
    }

}

File: StoreApi/Decorators/PizzaDecorator.cs
using StoreApi.Models;

namespace StoreApi.Decorators
{
    public abstract class PizzaDecorator : IPizza
    {
        protected IPizza _pizza;
        public PizzaDecorator(IPizza pizza)
        {
            _pizza = pizza;
        }

        public virtual string GetDescription() => "Plain Pizza";
        public virtual double GetCost() => 5.00; // Base cost for a plain pizza
    }
}
